File: CompassReports.Resources/Models/AssessmentFilterModel.cs
using System.Collections.Generic;

namespace CompassReports.Resources.Models
{
    public class AssessmentFilterModel
    {
        public List<int> Assessments { get; set; }
        public string AssessmentTitle { get; set; }
        public List<string> EnglishLanguageLearnerStatuses { get; set; }
        public List<string> Ethnicities { get; set; }
        public List<int> ExcludePerformanceKeys { get; set; }
        public List<int> PerformanceKeys { get; set; }
        public List<int> GoodCauseExcemptions { get; set; }
        public List<string> LunchStatuses { get; set; }
        public short SchoolYear { get; set; }
        public List<string> SpecialEducationStatuses { get; set; }
        public string Subject { get; set; }
    }
}

File: CompassReports.Web/App/Models/assessment-filter.model.ts
module App.Models {
    export class AssessmentFilterModel implements IReportFilterModel {
        SchoolYear?: number;
        SchoolYears?: number[];

        Assessments: number[];
        EnglishLanguageLearnerStatuses: string[];
        Ethnicities: string[];
        LunchStatuses: string[];
        SpecialEducationStatuses: string[];

        Schools: number[];
        Districts: number[];

        AssessmentTitle: string;
        Subject: string;
        PerformanceKey: number;

        addChartFilters = (chart: Models.ChartModel) => {
            if(chart.ChartFilters)
             this.PerformanceKey = chart.ChartFilters.PerformanceKey;
        }

        filteringCount = () => {
            //Starts at one since Grades always shows
            let count = 1;

            if (this.Subject) count++;
            if (this.SchoolYear != null) count++;
            if (this.EnglishLanguageLearnerStatuses != null && this.EnglishLanguageLearnerStatuses.length) count++;
            if (this.Ethnicities != null && this.Ethnicities.length) count++;
            if (this.SchoolYears != null && this.SchoolYears.length) count++;
            if (this.LunchStatuses != null && this.LunchStatuses.length) count++;
            if (this.SpecialEducationStatuses != null && this.SpecialEducationStatuses.length) count++;
            if (this.Schools != null && this.Schools.length) count++;
            if (this.Districts != null && this.Districts.length) count++;

            return count;
        }

        isFiltering = () => {
            return this.filteringCount() > 0;
        }

        reset = () => {
            this.Ethnicities = [];
            this.SchoolYears = [];
            this.EnglishLanguageLearnerStatuses = [];
            this.LunchStatuses = [];
            this.SpecialEducationStatuses = [];
            this.Assessments = [];
            this.SchoolYear = null;
            this.Subject = null;
        }

        constructor() {
            this.reset();
        }
    }
}